src/UserSpaceInstrumentation/Trampoline.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "Trampoline.h"
+
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_format.h>
+#include <absl/strings/str_split.h>
+#include <unistd.h>
+
+#include <cstdint>
+#include <string>
+
+#include "AllocateInTracee.h"
+#include "OrbitBase/ExecuteCommand.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+
+namespace orbit_user_space_instrumentation {
+
+using absl::numbers_internal::safe_strtou64_base;
+using orbit_base::ReadFileToString;
+
+bool DoAddressRangesOverlap(const AddressRange& a, const AddressRange& b) {
+  return !(b.end <= a.start || b.start >= a.end);
+}
+
+std::optional<size_t> LowestIntersectingAddressRange(const std::vector<AddressRange>& all_ranges,
+                                                     const AddressRange& range) {
+  for (size_t i = 0; i < all_ranges.size(); i++) {
+    if (DoAddressRangesOverlap(all_ranges[i], range)) {
+      return i;
+    }
+  }
+  return std::nullopt;
+}
+
+std::optional<size_t> HighestIntersectingAddressRange(const std::vector<AddressRange>& all_ranges,
+                                                      const AddressRange& range) {
+  for (int i = all_ranges.size() - 1; i >= 0; i--) {
+    if (DoAddressRangesOverlap(all_ranges[i], range)) {
+      return i;
+    }
+  }
+  return std::nullopt;
+}
```

Comment:
Yes, these ranges need to be sorted. And we only ever hand over sorted ranges since this is the way get them from GetTakenAddressRanges. 
So I'd rather make this explicit in the naming and documentation than adding a sort that would be an no-op.